api/graphql/types/helsinki_profile/types.py

```diff
@@ -0,0 +1,120 @@
+import graphene
+from graphene_django_extensions.errors import GQLNodePermissionDeniedError
+from graphql import GraphQLError
+from query_optimizer.selections import get_field_selections
+
+from api.graphql.extensions import error_codes
+from applications.models import Application
+from common.typing import GQLInfo
+from permissions.helpers import can_view_users
+from reservations.models import Reservation
+from users.helauth.clients import HelsinkiProfileClient
+from users.helauth.typing import LoginMethod, UserProfileInfo
+from users.models import User
+
+__all__ = [
+    "HelsinkiProfileDataNode",
+]
+
+
+class HelsinkiProfileDataNode(graphene.ObjectType):
+    first_name = graphene.String()
+    last_name = graphene.String()
+    email = graphene.String()
+    phone = graphene.String()
+    birthday = graphene.Date()
+    ssn = graphene.String()
+    street_address = graphene.String()
+    postal_code = graphene.String()
+    city = graphene.String()
+    municipality_code = graphene.String()
+    municipality_name = graphene.String()
+    login_method = graphene.Field(graphene.Enum.from_enum(LoginMethod))
+    is_strong_login = graphene.Boolean(required=True)
+
+    @classmethod
+    def get_data(
+        cls,
+        info: GQLInfo,
+        *,
+        application_id: int | None = None,
+        reservation_id: int | None = None,
+    ) -> UserProfileInfo:
+        if not can_view_users(info.context.user):
```

Comment:
Can't limit by `can_view_users()` since the _viewer_ and _handler_ roles do not have that permission; only by that have permission on the application or reservation.

Reply:
Oh, I forgot to change this.